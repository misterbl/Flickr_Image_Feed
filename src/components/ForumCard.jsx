import React from "react";

const ForumCard = ({ title, link, description, author, author_id }) => (
  <div className="ba pa3 ma4 overflow-hidden">
    <p>
      <a href={link}>{`${title}`}</a> by
      <a href={`https://www.flickr.com/photos/${author_id}`}>{` ${author}`}</a>
    </p>
    <p
      dangerouslySetInnerHTML={{ __html: `Description ${description}` }}
      className="mt4"
    />
  </div>
);
export default ForumCard;
