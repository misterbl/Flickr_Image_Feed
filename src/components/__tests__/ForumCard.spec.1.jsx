import * as React from "react";
import { shallow } from "enzyme";
import ForumCard from "../ForumCard";

describe("ForumCard", () => {
  const props = {
    title: "titleTest",
    author: "authorTest",
    author_id: "123Test",
    link: "linkTest",
    description: "descriptionTest"
  };

  const wrapper = shallow(<ForumCard {...props} />);
  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
