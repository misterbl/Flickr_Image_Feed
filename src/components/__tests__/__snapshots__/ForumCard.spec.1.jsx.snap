// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForumCard matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForumCard
    author="authorTest"
    author_id="123Test"
    description="descriptionTest"
    link="linkTest"
    title="titleTest"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          <a
            href="linkTest"
          >
            titleTest
          </a>
           by
          <a
            href="https://www.flickr.com/photos/123Test"
          >
             authorTest
          </a>
        </p>,
        <p
          className="mt4"
          dangerouslySetInnerHTML={
            Object {
              "__html": "Description descriptionTest",
            }
          }
        />,
      ],
      "className": "ba pa3 mw5 ma4 overflow-hidden",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              href="linkTest"
            >
              titleTest
            </a>,
            " by",
            <a
              href="https://www.flickr.com/photos/123Test"
            >
               authorTest
            </a>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "titleTest",
              "href": "linkTest",
            },
            "ref": null,
            "rendered": "titleTest",
            "type": "a",
          },
          " by",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " authorTest",
              "href": "https://www.flickr.com/photos/123Test",
            },
            "ref": null,
            "rendered": " authorTest",
            "type": "a",
          },
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "mt4",
          "dangerouslySetInnerHTML": Object {
            "__html": "Description descriptionTest",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            <a
              href="linkTest"
            >
              titleTest
            </a>
             by
            <a
              href="https://www.flickr.com/photos/123Test"
            >
               authorTest
            </a>
          </p>,
          <p
            className="mt4"
            dangerouslySetInnerHTML={
              Object {
                "__html": "Description descriptionTest",
              }
            }
          />,
        ],
        "className": "ba pa3 mw5 ma4 overflow-hidden",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                href="linkTest"
              >
                titleTest
              </a>,
              " by",
              <a
                href="https://www.flickr.com/photos/123Test"
              >
                 authorTest
              </a>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "titleTest",
                "href": "linkTest",
              },
              "ref": null,
              "rendered": "titleTest",
              "type": "a",
            },
            " by",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " authorTest",
                "href": "https://www.flickr.com/photos/123Test",
              },
              "ref": null,
              "rendered": " authorTest",
              "type": "a",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "mt4",
            "dangerouslySetInnerHTML": Object {
              "__html": "Description descriptionTest",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
